metaflow/metadata/service.py

```diff
@@ -41,7 +41,13 @@ def new_run_id(self, tags=[], sys_tags=[]):
         return self._new_run(tags=tags, sys_tags=sys_tags)
 
     def register_run_id(self, run_id, tags=[], sys_tags=[]):
-        pass
+        try:
```

Comment:
The comment and this code is confusing. I believe the idea is that if the run id is an integer, it was obtained from the service and doesn't need to be created whereas if the integer is a string, it was generated outside the service and needs to be registered. I would put a comment here and below around this. We should also ensure that IDs generated outside of the service are NEVER integers. Also, s/MLI/Metadata/

Reply:
Yes, currently we ensure that is the case for SFN by prepending SFN to the execution id.

Reply:
That's fine. I would rework the comment though to make it clear. It always confused me :).